proxy/ProxyTransaction.cc

```diff
@@ -240,3 +240,11 @@ ProxyTransaction::allow_half_open() const
 {
   return false;
 }
+
+// Most protocols will not want to set the Connection: header
+// For H2 it will initiate the drain logic.  So we make do nothing
+// the default action.
```

Comment:
For H2, I agree with we should not start the graceful shutdown. I'm a bit wondering about we should close the stream (send RST_STREAM frame to the peer).

Reply:
With the normal processing the stream will shutdown.  With HTTP/1.1 the response is sent as normal even it if has Connection: close.  With HTTP/2 after the response header and body are sent, the EOS bit will be sent on the last frame which will cause the stream to shutdown on the other side.

Reply:
OK, then, do nothing :+1: